nucypher/policy/conditions/jwt.py

```diff
@@ -0,0 +1,156 @@
+from typing import Any, Optional, Tuple
+
+import jwt
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric import ec, rsa
+from cryptography.hazmat.primitives.serialization import load_pem_public_key
+from marshmallow import ValidationError, fields, post_load, validate, validates
+
+from nucypher.policy.conditions.base import AccessControlCondition, ExecutionCall
+from nucypher.policy.conditions.context import (
+    is_context_variable,
+    resolve_any_context_variables,
+)
+from nucypher.policy.conditions.exceptions import InvalidCondition, JWTException
+from nucypher.policy.conditions.lingo import ConditionType
+from nucypher.utilities.logging import Logger
+
+
+class JWTVerificationCall(ExecutionCall):
+
+    _valid_jwt_algorithms = (
+        "ES256",
+        "RS256",
+    )  # https://datatracker.ietf.org/doc/html/rfc7518#section-3.1
+
+    SECP_CURVE_FOR_ES256 = "secp256r1"
+
+    class Schema(ExecutionCall.Schema):
+        jwt_token = fields.Str(required=True)
+        # TODO: See #3572 for a discussion about deprecating this in favour of the expected issuer
+        public_key = fields.Str(
+            required=True
+        )  # required? maybe a valid PK certificate passed by requester?
+        expected_issuer = fields.Str(required=False, allow_none=True)
+        # TODO: StringOrURI as per the spec.
+
+        @post_load
+        def make(self, data, **kwargs):
+            return JWTVerificationCall(**data)
+
+        @validates("jwt_token")
+        def validate_jwt_token(self, value):
+            if not is_context_variable(value):
+                raise ValidationError(
+                    f"Invalid value for JWT token; expected a context variable, but got '{value}'"
+                )
+
+        @validates("public_key")
+        def validate_public_key(self, value):
+            try:
+                public_key = load_pem_public_key(
+                    value.encode(), backend=default_backend()
+                )
+                if isinstance(public_key, rsa.RSAPublicKey):
+                    return value
```

Comment:
This function is "void", so you don't actually have to return anything. Either it raises an exception (failed validation) or not (all is well).

```suggestion
                    return
```

Reply:
2b048ea3ee847bb6a15494dae71a025ec744c3c2